sway/tree/workspace.c

```diff
@@ -202,14 +201,9 @@ char *workspace_next_name(const char *output_name) {
 	// As a fall back, get the current number of active workspaces
 	// and return that + 1 for the next workspace's name
 	int ws_num = root_container.children->length;
-	if (ws_num >= 10) {
-		l = 2;
-	} else if (ws_num >= 100) {
-		l = 3;
-	}
+	int l = snprintf(NULL, 0, "%d", ws_num);
 	char *name = malloc(l + 1);
-	if (!name) {
-		wlr_log(L_ERROR, "Could not allocate workspace name");
+	if (!sway_assert(name, "Cloud not allocate workspace name")) {
 		return NULL;
 	}
 	sprintf(name, "%d", ws_num++);
```

Comment:
`++` doesn't do anything useful here

Reply:
Well tbh relying on list length when there can be holes is pretty silly in the first place, a proper fix would be to walk over the list and look for the smallest number not taken (keep calling get_ws_by_name until none comes up?) or the highest number actually used (which could be much smaller than list length if they use names, and only cost trying to convert strings to ints), or it could be assuming the `0`th output is set and want to print `ws_num` with a useless `++` (ok that one is far fetched given default config sets workspace 10 on 0 key)

Tempted to call an out of scope on this one, just setting +1 doesn't fix much, and I'm too lazy to decide on what kind of algorithm to pick

Reply:
I agree, it's out of scope.